src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -182,7 +184,9 @@ module hybrid_ensemble_isotropic
   integer(r_kind) :: nval_loc_en
 
 ! For MGBF
+#ifdef USE_MGBF_def
```

Comment:
`USE_MGBF_def` is the **wrong** variable.  `CMakeLists.txt` sets `USE_MGBF`.  `USE_MGBF_def` needs to be changed to `USE_MGBF` 

The same comment applies to all occurrences of `USE_MGBF_def` in this source code file.

Reply:
I wanted to use different names in cmake files and the fortran source codes 
in src/gsi/CMakeLists.txt,  there is :
```
if(USE_MGBF)
  list(APPEND GSI_Fortran_defs USE_MGBF_def)
endif()
```
That makes USE_MGBF_def to be used as the preprocessing option.

Reply:
I set `USE_MGBF` to `ON` in `CMakeLists.txt`.  This broke the build
```
[ 80%] Building Fortran object src/gsi/CMakeFiles/gsi_fortran_obj.dir/hybrid_ensemble_isotropic.F90.o
/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/test/src/gsi/hybrid_ensemble_isotropic.F90(188): error #6457: This derived type name has not been declared.   [MG_INTSTATE_TYPE]
  type (mg_intstate_type), allocatable, dimension(:) :: obj_mgbf
--------^
/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/test/src/gsi/hybrid_ensemble_isotropic.F90(4184): error #6158: The structure-name is invalid or is missing.   [OBJ_MGBF]
  real(r_kind)   ,intent(inout) :: g(grd_loc%nsig,obj_mgbf(ig)%nm,obj_mgbf(ig)%mm)
--------------------------------------------------^
/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/test/src/gsi/hybrid_ensemble_isotropic.F90(4184): error #6158: The structure-name is invalid or is missing.   [OBJ_MGBF]
  real(r_kind)   ,intent(inout) :: g(grd_loc%nsig,obj_mgbf(ig)%nm,obj_mgbf(ig)%mm)
------------------------------------------------------------------^
/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/test/src/gsi/hybrid_ensemble_isotropic.F90(1769): error #6404: This name does not have a type, and must have an explicit type.   [PRINT_CPU]
       if(l_mgbf_loc) call print_mg_timers("mgbf_timing_cpu.csv", print_cpu, mype)
------------------------------------------------------------------^
/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/test/src/gsi/hybrid_ensemble_isotropic.F90(3749): error #6404: This name does not have a type, and must have an explicit type.   [OBJ_MGBF]
     allocate(work_mgbf(obj_mgbf(1)%km_a_all,obj_mgbf(1)%nm,obj_mgbf(1)%mm))
------------------------^
```

Looks like the error is that the first MGBF `ifdef` in `hybrid_ensemble_isotropic.F90` references `USE_MGBF` 
```
 ! For MGBF
#ifdef USE_MGBF
   use mg_intstate
   use mg_timers
 #endif /* End of USE_MGBF block */
```
while all the other ifdefs reference `USE_MGBF_def`.  We need to pick one.

My preference is to use `USE_MGBF` consistently through `CMakeLists.txt` and the source code.  That said, I am not a cmake expert.  Maybe the the two varaible approach of `USE_MGBF` & `USE_MGBF_def` is better.

Reply:
Russ,  Thanks for exploring on them.  I think use of  one variable or two variable is just a stylish problem. In this case, up to now, there are no actual differences.  Using two variables ( as for USE_GSDCLOUD vs RR_CLOUDANALYSIS)   just gives a reminder on how those directives in the source codes are being recognized by the compiler. 
If it is ok with you, I still prefer to following  the use of such a "two variable" approach  in the previous GSI setup as for the above GSDCLOUD management.   
If you think the "one variable" approach is better enough that we don't need use the same "two variable" approach in GSI for similar situations , I can change it later .

Reply:
My preference remains to use one variable, `USE_MGBF`, all the way through.  This is more readable than `USE_MGBF_def` and less confusing.  You are the developer.  It's your PR.  Do as you wish.